packages/next/build/webpack/plugins/middleware-plugin.ts

```diff
@@ -228,6 +230,36 @@ function isNodeJsModule(moduleName: string) {
   return require('module').builtinModules.includes(moduleName)
 }
 
+function getEdgeEntryBuildInfo(
+  moduleGraph: webpack.ModuleGraph,
+  module: webpack.Module
+) {
+  let currentModule: webpack.Module | null = module
+  while (currentModule?.layer === 'middleware') {
+    const buildInfo = getModuleBuildInfo(currentModule)
+    if (buildInfo.edgeFunctionConfig) {
+      return buildInfo
+    }
+    currentModule = moduleGraph.getIssuer(currentModule)
```

Comment:
This is only one (of potential multiple) importer path. One module might be used by multiple entrypoints with different config. You must delay error reporting until the full module graph is done.